src/blockdev.rs

```diff
@@ -893,19 +915,31 @@ fn blkid() -> Result<Vec<HashMap<String, String>>> {
     // the blkid -p below, which we use to probe the devices to not rely on
     // the blkid cache:
     // https://github.com/coreos/fedora-coreos-config/pull/2181#issuecomment-1397386896
-    let devices = {
-        let mut cmd = Command::new("blkid");
-        cmd.arg("--cache-file");
-        cmd.arg("/dev/null");
-        cmd.arg("-o");
-        cmd.arg("device");
-        cmd_output(&mut cmd)?
-    };
+    let mut cmd_dev: Vec<&Path> = Vec::new();
+    let devices;
+    if dev.is_none() {
+        devices = {
+            let mut cmd = Command::new("blkid");
+            cmd.arg("--cache-file");
+            cmd.arg("/dev/null");
+            cmd.arg("-o");
+            cmd.arg("device");
+            cmd_output(&mut cmd)?
+        };
 
+        cmd_dev.extend(devices
+            .lines()
+            .filter_map(|line|Some(Path::new(line.trim())))
+        );
+    } else {
+        if let Some(path) = dev {
+            cmd_dev.extend(path);
+        }
+    }
```

Comment:
A more idiomatic way to restructure all this is:

```rust
let cmd_dev = if let Some(paths) = dev {
    paths
} else {
    let devices = {
        ...
    };
    devices.lines().map(|line| Path::new(line.trim())).collect()
};
```

Reply:
Thanks for the free rust class @jlebon ;)